safe_transaction_service/history/services/transaction_service.py

```diff
@@ -371,3 +373,546 @@ def serialize_all_txs_v2(
 
         logger.debug("Serialized all transactions")
         return results
+
+    def get_export_transactions(
+        self,
+        safe_address: str,
+        execution_date_gte: Optional[datetime] = None,
+        execution_date_lte: Optional[datetime] = None,
+        limit: int = 1000,
+        offset: int = 0,
+    ) -> Tuple[List[Dict[str, Any]], int]:
+        """
+        Get transactions optimized for CSV export using raw SQL queries
+
+        :param safe_address: Safe address to get transactions for
+        :param execution_date_gte: Filter transactions executed after this date
+        :param execution_date_lte: Filter transactions executed before this date
+        :param limit: Maximum number of transactions to return
+        :param offset: Number of transactions to skip
+        :return: Tuple of (transactions, total_count)
+        """
+        logger.debug(
+            "[%s] Getting export transactions with raw SQL: gte=%s, lte=%s, limit=%d, offset=%d",
+            safe_address,
+            execution_date_gte,
+            execution_date_lte,
+            limit,
+            offset,
+        )
+
+        # Base WHERE conditions for the final SELECT
+        where_conditions = []
+        params = []
+
+        if execution_date_gte:
+            where_conditions.append("execution_date >= %s")
+            params.append(execution_date_gte)
+        if execution_date_lte:
+            where_conditions.append("execution_date <= %s")
+            params.append(execution_date_lte)
+
+        where_clause = " AND ".join(where_conditions) if where_conditions else "1=1"
+
+        # Main query that unions all transaction types with their transfers
+        main_query = f"""
```

Comment:
Based on our experience with `all-transactions` endpoint, this query will be really slow to the point it will not end for some big Safes. The `relevant txs` table needs to be used to filter, and then join the required tables to get the information (take a look at the `all-transactions` endpoint)

The query needs to be tested in our staging database

Reply:
Makes sense, yes